package/MDAnalysis/topology/DMSParser.py

```diff
@@ -176,21 +176,43 @@ def dict_factory(cursor, row):
         topattrs.append(Atomtypes(atomtypes, guessed=True))
 
         # Residues
-        atom_residx, res_resids, (res_resnames, res_segids) = squash_by(
-            attrs['resid'], attrs['resname'], attrs['segid'])
+        atom_residx, (res_resids,
+                      res_resnums,
+                      res_resnames,
+                      res_segids) = change_squash(
+            (attrs['resid'], attrs['resname'], attrs['segid']),
+            (attrs['resid'],
+             attrs['resid'].copy(),
+             attrs['resname'],
+             attrs['segid']),
+            )
+
+        n_residues = len(res_resids)
         topattrs.append(Resids(res_resids))
-        topattrs.append(Resnums(res_resids.copy()))
+        topattrs.append(Resnums(res_resnums))
         topattrs.append(Resnames(res_resnames))
 
-        # Segments
-        res_segidx, seg_segids = squash_by(
-            res_segids)[:2]
-        topattrs.append(Segids(seg_segids))
+        if any(res_segids) and not any(val is None for val in res_segids):
+            res_segidx, (res_segids,) = change_squash((res_segids,),
+                                                      (res_segids,))
+
+            uniq_seg = np.unique(res_segids)
```

Comment:
Are we normally using our own `lib.util.unique_int_1d` https://github.com/MDAnalysis/mdanalysis/blob/2550d061a3134e0002fc8ec4eb31eb3c63bcfa5b/package/MDAnalysis/lib/_cutil.pyx#L52 instead of `np.unique` for performance reasons, @richardjgowers @jbarnoud @tylerjereddy ?

Or will it not matter here because `res_segids` is small?

Reply:
res_segids should be pretty small most of the time. The only time I could see it becoming a problem is if for whatever reason the DMS was structures so that similar lipids/solvent molecules were not grouped in the database. It would have to be extremely disorganized to cause performance issues I would think.

Reply:
Ok, then I think we just leave it.

Reply:
Yeah performance, I wanted to get rid of our own version because it required maintenance to make 32-bit compatible but I think @lilyminium pointed out that it was a bit faster I guess. The NumPy one has quite a few additional features so that's probably part of the difference.